clang/include/clang/Serialization/ASTRecordReader.h

```diff
@@ -337,6 +337,14 @@ class ASTRecordReader
   /// Reads attributes from the current stream position, advancing Idx.
   void readAttributes(AttrVec &Attrs);
 
+  /// Reads one attribute from the current stream position, advancing Idx.
+  /// Parent Decl is provided to delay attribute deserialization.
```

Comment:
The lazy deserialization is a subtle difference, could we name the functions appropriately and "scary", e.g. `readAttrOrDelay`.